File: code/R/01_build_panel.R

library(tidyverse)
dir.create("data/clean", showWarnings = FALSE, recursive = TRUE)
panel <- tibble(
  country = character(),
  quarter = character(),
  capability_area = character(),
  shock_event = integer(),
  shock_severity = integer(),
  pre_shock_us_exposure = double(),
  pesco_projects = integer(),
  edf_awards_count = integer(),
  collab_proc_share = double(),
  eu_origin_proc_share = double(),
  autonomy_rhetoric_index = double(),
  threat_alignment = double(),
  defence_spend_gdp = double(),
  industry_capacity = double()
)
write_csv(panel, "data/clean/country_quarter_panel.csv")
message("Wrote data/clean/country_quarter_panel.csv")

File: code/python/01_build_panel.py

import pandas as pd
from pathlib import Path

RAW = Path("data/raw")
CLEAN = Path("data/clean")
CLEAN.mkdir(parents=True, exist_ok=True)

def main():
    # TODO: load raw sources (SIPRI/EDA/DSCA/GAO), harmonize to country-quarter
    # Example scaffold only:
    panel = pd.DataFrame(columns=[
        "country","quarter","capability_area","shock_event","shock_severity",
        "pre_shock_us_exposure","pesco_projects","edf_awards_count","collab_proc_share",
        "eu_origin_proc_share","autonomy_rhetoric_index","threat_alignment",
        "defence_spend_gdp","industry_capacity"
    ])
    out = CLEAN / "country_quarter_panel.csv"
    panel.to_csv(out, index=False)
    print(f"Wrote {out}")

if __name__ == "__main__":
    main()
